rustup-setup.sh

```diff
@@ -0,0 +1,211 @@
+#!/bin/sh
+# Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution and at
+# http://rust-lang.org/COPYRIGHT.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+# This is just a little script that can be curled from the internet to
+# install rustup. It just does platform detection, curls the installer
+# and runs it.
+
+set -u
+
+RUSTUP_UPDATE_ROOT="https://github.com/rust-lang-nursery/multirust-rs-binaries/raw/master"
+
+main() {
+    need_cmd curl
+    need_cmd mktemp
+    need_cmd chmod
+    need_cmd mkdir
+    need_cmd rm
+    need_cmd rmdir
+    need_cmd printf
+
+    get_architecture || return 1
+    local _arch="$RETVAL"
+    assert_nz "$_arch" "arch"
+
+    local _ext=""
+    case "$_arch" in
+	*windows*)
+	    _ext=".exe"
+	    ;;
+    esac
+
+    local _url="$RUSTUP_UPDATE_ROOT/$_arch/multirust-setup$_ext"
+
+    local _dir="$(ensure mktemp -d)"
+    local _file="$_dir/rustup-setup$_ext"
+
+    if [ -e /usr/bin/printf ]; then
+	# printf is a shell builtin and won't interpret \x in posix
+	# (sh) mode, so using the actual printf exe if it exists
+	/usr/bin/printf "\x1b[36;1minfo\x1b[0m: downloading installer\n"
```

Comment:
This prints `info` in cyan to match the exe's behavior and make the output seamless.